tsl/src/continuous_aggs/create.c

```diff
@@ -1855,12 +1856,23 @@ cagg_update_view_definition(ContinuousAgg *agg, Hypertable *mat_ht,
 	 */
 	Assert(list_length(view_query->targetList) == list_length(user_query->targetList));
 
+	TupleDesc desc = RelationGetDescr(user_view_rel);
+	int i = 0;
 	forboth (lc1, view_query->targetList, lc2, user_query->targetList)
 	{
 		TargetEntry *view_tle, *user_tle;
+		FormData_pg_attribute *attr = TupleDescAttr(desc, i);
 		view_tle = lfirst_node(TargetEntry, lc1);
 		user_tle = lfirst_node(TargetEntry, lc2);
-		view_tle->resname = user_tle->resname;
+		ereport(LOG,
```

Comment:
I think you added this for testing. We do not need a LOG message here. 
We need a test case for this change.
Would also help to have a comment in the code explaining why we have to match the names using the user_view_rel  and not the query. It is not obvious why there would be a discrepancy.

Reply:
Log message removed. Test case added (also an update test, which is what the original bug report was about). Tried to add an explanation.